app/models/spree_avatax/sales_invoice.rb

```diff
@@ -25,6 +25,11 @@ def generate(order)
 
       return if order.completed? || !SpreeAvatax::Shared.taxable_order?(order)
 
+      taxable_records = order.line_items + order.shipments
+      taxable_records.each do |taxable_record|
+        taxable_record.update_column(:pre_tax_amount, taxable_record.discounted_amount.round(2))
+      end
```

Comment:
@gmacdougall it does a whole bunch more than this: https://github.com/solidusio/solidus_avatax/blob/647b1bf749868a8ba9a73c1338cc1c4599805d93/app/models/spree_avatax/sales_shared.rb#L32-L52

Are you saying we should have more comments on the `generate` method? Or that a more descriptive name would be nice? Or that you wish we had a better way to integrate w/ taxes in Solidus? Or something else?